src/main/java/utils/FormUrlEncodedParsingStrategy.java

```diff
@@ -0,0 +1,28 @@
+package utils;
+
+import static constant.Constant.*;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class FormUrlEncodedParsingStrategy implements HttpBodyParsingStrategy {
+    @Override
+    public Map<String, String> parse(String body) {
+        if (body.isBlank()) {
+            return Map.of();
+        }
+
+        Map<String, String> map;
+        try {
+            map = Arrays.stream(body.split(PARAMETER_SEPARATOR))
+                    .map(s -> s.split(PARAMETER_EQUAL_SIGN))
+                    .map(arr -> Map.entry(arr[0], arr[1]))
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        } catch (Exception e) {
+            throw new IllegalArgumentException("유효하지 않은 형식입니다.");
+        }
+
+        return map;
```

Comment:
try 블록 내부에서 return을 해준다면  try 블록 밖에서 map 변수를 선언할 필요가 없겠네요. 🙂 
(선언과 동시에 할당을 수행할 수 있겠네요. )

그런데 try 블록 안에서 바로 리턴하게된다면 애초에 변수 선언이 불필요할 것 같아요. 😃

```suggestion
        try {
            return Arrays.stream(body.split(PARAMETER_SEPARATOR))
                    .map(s -> s.split(PARAMETER_EQUAL_SIGN))
                    .map(arr -> Map.entry(arr[0], arr[1]))
                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
        } catch (Exception e) {
            throw new IllegalArgumentException("유효하지 않은 형식입니다.");
        }
```

이 부분에서 조금 더 고민해볼 부분은 `PARAMETER_SEPARATOR`로 추출한 각 **요청 파라미터 key, value 라인**이 유효하지 않은 경우에 대해서도 처리해주면 어떨까요? 😃

Reply:
조금 고민을 해봤는데, header의 경우는 몇가지 제약사항이 있어 검증하는 로직을 집어넣었지만
body 내부의 값에 대해서는 잘 모르겠습니다. 어떤 제약 조건이 있을까요???

Reply:
제가 조금 모호하게 말씀을 드린 것 같네요. 🙂
key와 value가 한 쌍을 이루지 않은 케이스는 예외를 던진다거나하는 유효성 검증 로직을 넣어보면 좋을 것 같다는 이야기였어요. 🙂

Reply:
쌍이 이루어지지 않거나 할 때 split + 직접 접근을 하다보니 일단은 ArrayIndexOutOfBoundsException을 발생하는 것을 확인 했습니다!
나중에 시간이 남을때 추가적인 검증 로직을 추가하겠습니다!